### ISSUE TITLE
Regression: New LOGICAL_DNS clusters do not populate

### ISSUE THREAD

**Issue by Stono** [istio/istio#54062]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

This appears to be a regression of https://github.com/istio/istio/issues/49315

`LOGICAL_DNS` clusters on new deployments for me aren't working.

The error in pilot is:
```
2024-11-26T12:30:48.088386056Z {"level":"info","time":"2024-11-26T12:30:48.088207Z","scope":"ads","msg":"Push Status: {\n    \"pilot_dns_cluster_without_endpoints\": {\n        \"outbound|443||api.checklyhq.com\": {\n            \"proxy\": \"platform-checkly-agent-56f9d7bc49-w6vhb.platform-checkly-agent\",\n            \"message\": \"LOGICAL_DNS cluster without endpoints outbound|443||api.checklyhq.com found while pushing CDS\"\n        },\n        \"outbound|80||api.checklyhq.com\": {\n            \"proxy\": \"platform-checkly-agent-56f9d7bc49-w6vhb.platform-checkly-agent\",\n            \"message\": \"LOGICAL_DNS cluster without endpoints outbound|80||api.checklyhq.com found while pushing CDS\"\n        }\n    },\n    \"pilot_eds_no_instances\": {\n        \"outbound|443||api.checklyhq.com\": {},\n        \"outbound|80||api.checklyhq.com\": {}\n    }\n}"}
```

And as per last time, switching the `ServiceEntry` from `DNS_ROUND_ROBIN` to `DNS` and back "fixes" it.

cc @howardjohn @ericvn @ram as you worked on the original

### Version

```Text
1.23.1
```

**Comment by hzxuzhonghu** [comment]
How can you reproduce this error

**Comment by Stono** [comment]
Yeah it was, that's why I think it's regressed.  The same steps reproduce it.

**Comment by hzxuzhonghu** [comment]
I cannot reproduce on my istio 1.24.1 cluster

**Comment by Stono** [comment]
Interesting, I'm on 1.23; I have a 1.24 cluster in testing, let me see if the same happens for me

**Comment by Stono** [comment]
OK today I can't reproduce it on 1.23 or 1.24, something odd is going on.  Leave it with me and i'll try and sus it out.

**Comment by Stono** [comment]
This happened again today when I deployed a service which rapidly:
- deleted the associated resources
- created them again

but i wasn't able to recreate it when i tried.

As i've been burned a few times now by LOGICAL_DNS, i think i'm going to revert back to DNS_ROUND_ROBIN

**Comment by keithmattix** [comment]
And this happened on 1.24.1?

**Comment by Stono** [comment]
Aye :'( 

It's annoying me because I cannot find a reproduction!

**Comment by Evesy** [comment]
I've managed to reproduce the issue many times following the below steps; although there are near-equal number of times the below doesn't manifest in any problems, so it definitely feels like a race somewhere

Given a workload with the below `ServiceEntry` present:
```
apiVersion: networking.istio.io/v1
kind: ServiceEntry
metadata:
  name: httpbin
spec:
  exportTo:
  - .
  hosts:
  - httpbin.org
  location: MESH_EXTERNAL
  ports:
  - name: http-port
    number: 80
    protocol: HTTP
    targetPort: 443
  - name: https-port
    number: 443
    protocol: HTTPS
  resolution: DNS_ROUND_ROBIN
```

The workload has clusters for `httpbin.org`:
```
$ istioctl proxy-config cluster -n monkeynetes monkeynetes-86f5f8bf49-m7w5v | grep httpbin
httpbin.org                                      80       -          outbound      LOGICAL_DNS      <unknown>
httpbin.org                                      443      -          outbound      LOGICAL_DNS      <unknown>
```

If I create an identical `ServiceEntry` to the above only changing `metadata.name` to `httpbin-new` and run the below (applying the new resource, and deleting the existing one in quick succession), it often results in the workload being without any clusters for `httpbin.org`:

```
kubectl apply -f serviceEntryNewName.yaml && kubectl delete serviceentry httpbin
```

```
$ istioctl proxy-config cluster -n monkeynetes monkeynetes-86f5f8bf49-m7w5v | grep httpbin

```

I have also seen occasions where only one of the two expected clusters (two ports in the SE definition) ends up existing:
```
$ istioctl proxy-config cluster -n monkeynetes monkeynetes-86f5f8bf49-m7w5v | grep httpbin
httpbin.org                                      443      -          outbound      LOGICAL_DNS      <unknown>
```

**Comment by ramaraochavali** [comment]
I will debug this more. Assigning to me
/assign

**Comment by ramaraochavali** [comment]
@Evesy I was n't able to reproduce this locally. But I suspect based on this comment "I have also seen occasions where only one of the two expected clusters (two ports in the SE definition) ends up existing:" - it has some thing to do with DeltaXds. Is it possible to disable deltaXds (https://github.com/istio/istio/blob/master/releasenotes/notes/delta-xds-default.yaml) and rerun your experiment? If you are able to reproduce, please share istiod debug logs
